aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -793,13 +793,9 @@ _flash_attention_forward(
   // We are going to have two paths:
   // 1. The standard MHA path for dense tensors
   // 2. The Varseqlen path
-  TORCH_CHECK(
-      cumulative_sequence_length_q.has_value() ==
-          cumulative_sequence_length_k.has_value(),
-      "cumulative_sequence_length_q and cumulative_sequence_length_k must be both set or both not set");
   Tensor output, q_padded, k_padded, v_padded, logsumexp, output_shape,
       philox_seed, philox_offset, debug_attn_mask;
-  if (cumulative_sequence_length_q.has_value()) {
+  if (cumulative_sequence_length_q.defined() && cumulative_sequence_length_q.numel() > 0) {
```

Comment:
I did this trick somehwere else too right, I think it was because autograd turns Nonees into undefined tensors